### PR TITLE
Globally clip gradient norms by specified maximum norm correctly

### PR DESCRIPTION
**Pull request by amorehead**
* Globally clips gradient norms by a specified maximum norm value correctly using rank syncing
* See https://docs.pytorch.org/docs/stable/fsdp.html#torch.distributed.fsdp.FullyShardedDataParallel.clip_grad_norm_

### CODE REVIEW

**Comment by amorehead** [comment]
Also, on a different note, there's an MIT license statement at the bottom of the `README.md`, but the actual license attached to this repository is an Apache-2 license?

**Comment by amorehead** [comment]
Lastly, does [this line](https://github.com/amorehead/ZClip/blob/bcdab02154e74e66779d75d3a74d75914e65aa03/zclip.py#L177) accidentally accumulate the total gradient norm for each (e.g., GPU) rank instead of once for all ranks?

**Comment by amorehead** [comment]
Moving these changes over to [this PR](https://github.com/bluorion-com/ZClip/pull/3).
